src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+
+export const TodoModal = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodo) {
+      getUser(currentTodo.userId)
```

Comment:
Ensure that the `getUser` function correctly fetches user data based on `currentTodo.userId`. Handle potential errors in the promise chain to avoid runtime issues.